packages/markdown-docx/src/ToOOXMLVisitor/index.js

```diff
@@ -102,23 +126,55 @@ class ToOOXMLVisitor {
         return tag;
     }
 
+    /**
+     * Checks if an object contains the property and updates counter for it.
+     *
+     * @param {string} tag  Tag of the property
+     * @param {string} type Type of the property
+     */
+    createOrUpdateCounter(tag, type) {
+        if (Object.prototype.hasOwnProperty.call(this.counter, tag)) {
+            this.counter = {
+                ...this.counter,
+                [tag]: {
+                    ...this.counter[tag],
+                    count: ++this.counter[tag].count,
+                },
+            };
+        } else {
+            this.counter[tag] = {
+                count: 1,
+                type,
+            };
+        }
+    }
+
     /**
      * Traverses CiceroMark nodes in a DFS approach
      *
-     * @param {object} node       CiceroMark Node
-     * @param {array}  properties Properties to be applied on current node
+     * @param {object}  node                           CiceroMark Node
+     * @param {array}   properties                     Properties to be applied on current node
+     * @param {boolean} isConditionalOrOptionalPresent True if the dfs traversal contains conditional or optional else false
      */
-    traverseNodes(node, properties = []) {
+    traverseNodes(node, properties = [], isConditionalOrOptionalPresent = '') {
```

Comment:
The docstrings say it should be boolean however you have assigned it an empty string.

Reply:
I don't think so this parameter should be even present here given the function name. `traverseNodes` is such a generic name and it makes sense that it would have `node` and `properties` but `isConditonalOrOptional` just takes away is "genericity".

I hope this is resolved by the pseudocode I suggested.